membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -297,13 +350,238 @@ def test_analysis_mean(self, universe):
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap_xy,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin,
+                               wrap=False).run()
+        avg_surface = mc.results.average_z_surface
+        assert_almost_equal(avg_surface, expected_surface)
+
+    # test mean
+    def test_analysis_mean_no_wrap(self, universe_dummy_wrap):
+        expected_mean = np.array(np.full((3, 3), np.nan))
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin,
+                               wrap=False).run()
+        avg_mean = mc.results.average_mean
+        assert_almost_equal(avg_mean, expected_mean)
+
+    def test_analysis_mean_no_wrap(self, universe_dummy_wrap_xy):
+        expected_mean = np.array(np.full((3, 3), np.nan))
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap_xy,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin,
+                               wrap=False).run()
+        avg_mean = mc.results.average_mean
+        assert_almost_equal(avg_mean, expected_mean)
+
+    # test gaussian
+    def test_analysis_gaussian_no_wrap(self, universe_dummy_wrap):
+        expected_gaussian = np.array(np.full((3, 3), np.nan))
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin,
+                               wrap=False).run()
+        avg_gaussian = mc.results.average_gaussian
+        assert_almost_equal(avg_gaussian, expected_gaussian)
+
+    def test_analysis_gaussian_no_wrap(self, universe_dummy_wrap_xy):
+        expected_gaussian = np.array(np.full((3, 3), np.nan))
+        x_bin = y_bin = 3
+        mc = MembraneCurvature(universe_dummy_wrap_xy,
+                               n_x_bins=x_bin,
+                               n_y_bins=y_bin,
+                               wrap=False).run()
+        avg_gaussian = mc.results.average_gaussian
+        assert_almost_equal(avg_gaussian, expected_gaussian)
+
+    @pytest.mark.parametrize('x_bin, y_bin, expected_surface', [
+        (3, 3,
+         np.array([[150., 120., 150.],
+                   [150., 120., 120.],
+                   [150., 120., 120.]])),
+        (4, 4,
+         np.array([[150., 120., 150., np.nan],
+                   [150., 120., 120., np.nan],
+                   [150., 120., 120., np.nan],
+                   [np.nan, np.nan, np.nan, np.nan]])),
+        (5, 5,
+         np.array([[150., 120., np.nan, 150., np.nan],
+                  [150., 120., np.nan, 120., np.nan],
+                  [np.nan, np.nan, np.nan, np.nan, np.nan],
+                  [150., 120., np.nan, 120., np.nan],
+                  [np.nan, np.nan, np.nan, np.nan, np.nan]]))
```

Comment:
Why are the nans now in the 3rd row and column? Should it not be like the 4x4 grid but with nans in the extra rows and columns? I don't think this is the answer we'd want to get for the surface.

Reply:
Ah, I forgot that xrange and yrange defaulted to the universe dimensions. Never mind.

Reply:
This test runs with the grofile with 9 particles (`GRO_PO4_SMALL`). In the file, the 9 particles in the system are distributed like this:
```
   +-----------+
   | 7 | 8 | 9 | 
   +-----------+
   | 4 | 5 | 6 |
   +-----------+
   | 1 | 2 | 3 | 
   +-----------+
```
, where atoms 1-4 have `z=150`, and atoms 5-9 have `z=120`. 
Then, when we divide the same grid into 4 bins, we have that each bin covers a region of 0.75

```
  y = 3+---------------+
       | x | x | x | x | 
       +---------------+
       | 7 | 8 | 9 | x | 
       +---------------+
       | 4 | 5 | 6 | x | 
       +---------------+
       | 1 | 2 | 3 | x |
 y = 0 +---------------+
      +---|---|---|----+ 
       x=0             x=3
```
With 5 bins, we have that each bin covers a region of 0.6. 
Atom 3= [ 20.,   0., 150.], --> `1.8 < x < 2.4`, `0. < y < 0.6`,
Atom 6= [ 20.,   10., 150.], --> `1.8 < x < 2.4`, `0.6 < y < 1.2`
Atom 7 = [  0.,  20., 120.], --> `0. < x < 0.6`, `1.8 < y < 2.4`
Atom 8=  [ 10.,  20., 120.], --> `0.6 < x < 1.2`, `1.8 < y < 2.4`
Atom 9 = [ 20.,  20., 120.] --> `1.8 < x < 2.4`, `1.8 < y < 2.4`
then all the space between `1.8 < x < 2.4` and 1.8 < y < 2.4` is empty.
```
y=3 +-------------------+
    | x | x | x | x | x | 
2.4 +-------------------+
    | 7 | 8 | x | 9 | x | 
1.8 +-------------------+
    | x | x | x | x | x |
1.2 +-------------------+
    | 4 | 5 | x | 6 | x |
0.6 +-------------------+
    | 1 | 2 | x | 3 | x |
y=0 +-------------------+
    +---|---|---|---|---+ 
   x=0 0.6 1.2 1.8 2.4 x=3
```
This is why the 3rd row and column of nans. Sorry it took forever to answer.